src/main/java/com/bench/stockmanagement/ProductHandler.java

```diff
@@ -1,35 +1,122 @@
+import com.bench.stockmanagement.domain.Order;
 import com.bench.stockmanagement.domain.Product;
+import com.bench.stockmanagement.domain.ProductStockData;
+import com.bench.stockmanagement.domain.Result;
 import com.bench.stockmanagement.mappers.ProductMapper;
 import com.bench.stockmanagement.services.OrderReader;
-import com.bench.stockmanagement.services.SellingReader;
-import com.bench.stockmanagement.services.dynamo.OrderService;
-import com.bench.stockmanagement.services.dynamo.SellingService;
+import com.bench.stockmanagement.services.dynamo.OrderRepository;
+import com.bench.stockmanagement.services.dynamo.ProductRepository;
+import com.bench.stockmanagement.services.dynamo.SellingRepository;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import software.amazon.awssdk.core.async.SdkPublisher;
+import software.amazon.awssdk.enhanced.dynamodb.model.Page;
 
 import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.bench.stockmanagement.domain.Result.FAIL;
+import static com.bench.stockmanagement.domain.Result.SUCCESS;
 
 @Component
 public class ProductHandler {
+    private final OrderReader reader;
     private final ProductMapper productMapper;
-    private final OrderService orderService;
-    private final SellingService sellingService;
+    private final OrderRepository orderRepository;
+    private final SellingRepository sellingRepository;
+    private final ProductRepository productRepository;
 
     @Autowired
-    public ProductHandler(ProductMapper productMapper, OrderService orderService, SellingService sellingService) {
+    public ProductHandler(OrderReader reader, ProductMapper productMapper, OrderRepository orderRepository,
+            SellingRepository sellingRepository, ProductRepository productRepository) {
+        this.reader = reader;
         this.productMapper = productMapper;
-        this.orderService = orderService;
-        this.sellingService = sellingService;
+        this.orderRepository = orderRepository;
+        this.sellingRepository = sellingRepository;
+        this.productRepository = productRepository;
+    }
+
+    // Save products
+    public Mono<Result> saveProducts() {
+        List<Order> orders = reader.readOrder();
+        List<DBProduct> dbProducts = productMapper.mapProduct(orders);
+
+        return Flux.fromIterable(dbProducts).log()
+                .doOnNext(productRepository::saveProducts)
+                .then(Mono.just(SUCCESS))
+                .onErrorReturn(FAIL);
+    }
+
+    // Get product by itemNumber
+    // Item number, how many is expected in stock min/max, purchase prices
+    public Mono<List<ProductStockData>> getProductByItemNumber(String itemNumber) {
+        return Mono.fromFuture(productRepository.getProductByItemNumber(itemNumber))
```

Comment:
It's better to use the lambda version of fromFuture:
```suggestion
        return Mono.fromFuture(() -> productRepository.getProductByItemNumber(itemNumber))
```
While CompletableFuture is eager (executed as soon as it called), Mono is lazy (nothing happens until subscribed). To resolve this difference, the lambda version of this method will delay calling the future until subscription.